contracts/ConverterRamp.sol

```diff
@@ -204,11 +204,15 @@ contract ConverterRamp is Ownable {
         TokenConverter tokenConverter = TokenConverter(_converter);
 
         /// store the previus balance after conversion to validate
-        uint256 prevBalance = address(this).balance;
+        uint256 prevEthBal = address(this).balance - msg.value;
 
         /// call convert in token converter
         tokenConverter.convert.value(_amount)(fromToken, toToken, _amount, _tokenCost, _etherCost, msg.sender);
 
+        /// give back the surplus if the sender send more eth
```

Comment:
this logic is duplicate in the proxy, but this proposal i like. it would be necessary to eliminate this logic from the proxy

Reply:
https://github.com/ripio/ramp-contracts/pull/9